include/aspect/material_model/interface.h

```diff
@@ -775,8 +775,41 @@ namespace aspect
         * transport) in each quadrature point.
         */
         std::vector<double> rhs_melt_pc;
+
     };
 
+    /**
+     * A class for an elastic force term to be added to the RHS of the
+     * Stokes system, which can be attached to the
+     * MaterialModel::MaterialModelOutputs structure and filled in the
+     * MaterialModel::Interface::evaluate() function.
+     */
+    template <int dim>
+    class ElasticOutputs: public AdditionalMaterialOutputs<dim>
+    {
+      public:
+        ElasticOutputs(const unsigned int n_points)
+          : rhs_e(n_points)
```

Comment:
@gassmoeller - I'm struggling a bit on this point. I've tried a number a permutations for initiating `rhs_e` (now renamed to `elastic_force`) with signaling NaNs and continue to get compiler errors.

I thought `elastic_force(n_points, numbers::signaling_nan<std::vector<Tensor<2,dim> > >() )` would be the correct structure, but apparently not. Before passing on compiler errors, is the structure even in the ballpark of what you had in mind?

Reply:
You are close. I think it should be `elastic_force(n_points, numbers::signaling_nan<Tensor<2,dim> >() )` The first parameter is the number of entries, the second parameter is the value that each entry will be initialized to.

Reply:
Doh. Thanks for the fix (compiles successfully now)! 

The bad news is this leads to a segmentation in the stokes assembler during the first time step, when these values have not been calculated yet. Should I put a flag in the assembler to not due this operation on the first time step? Alternatively, I can simply pass a zero value during the initial time step. 

In some studies people often prescribe an initial stress state, so perhaps this second option is the best way forward in terms of future compatibility?

Reply:
Yes, I think you should decide what these values should be in the initial time step (probably either zero or the initial stress state) and then fill them accordingly in time step zero. The problem occurs because in line 264 of `viscoelastic.cc`, you only fill the elastic outputs if the time step is larger than 0, so you just have to add a separate if that says how the values should be filled if the time step is zero.

Reply:
:+1: